service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java

```diff
@@ -151,96 +149,87 @@ public GetTariffServiceDto editTariffService(EditTariffServiceDto dto, Integer i
     }
 
     @Override
-    public CreateServiceDto addService(CreateServiceDto dto, String uuid) {
-        User user = userRepository.findByUuid(uuid);
-        Service service = createServiceWithTranslation(dto, user);
-        service.setFullPrice(getFullPrice(dto.getPrice(), dto.getCommission()));
-        serviceRepository.save(service);
-        serviceTranslationRepository.saveAll(service.getServiceTranslations());
-        return modelMapper.map(service, CreateServiceDto.class);
+    public GetServiceDto addService(CreateServiceDto dto, String uuid) {
+        Service service = serviceRepository.save(createService(dto, uuid));
+        return modelMapper.map(service, GetServiceDto.class);
+    }
+
+    private Service createService(CreateServiceDto dto, String uuid) {
+        long tariffId = dto.getTariffId();
+        Employee employee = getEmployeeByUuid(uuid);
+        TariffsInfo tariffsInfo = getTariffsInfoById(tariffId);
+        Optional<Service> service = getServiceByTariffsInfoId(tariffId);
+        if (service.isEmpty()) {
```

Comment:
you can call 
```
        Employee employee = getEmployeeByUuid(uuid);
        TariffsInfo tariffsInfo = getTariffsInfoById(tariffId);
```
only when you are sure that there is no service

Reply:
changed